appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java

```diff
@@ -179,6 +180,15 @@
 		}
 	}
 
+	private static void assertValidDeviceDataForRegister(DeviceRegistrationRequestDto requestDto)
+	{
+		String hint = "All properties are manadatory, except for the FireBase ID";
+		Require.isNonNull(requestDto.name, () -> InvalidDataException.missingProperty("name", hint));
+		Require.isNonNull(requestDto.operatingSystemStr, () -> InvalidDataException.missingProperty("operatingSystem", hint));
+		Require.isNonNull(requestDto.appVersion, () -> InvalidDataException.missingProperty("appVersion", hint));
+		Require.that(requestDto.appVersionCode != 0, () -> InvalidDataException.missingProperty("appVersionCode", hint));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "appVersionCode" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1192)

Reply:
I'm ignoring this one, as the code is more clear with the property name spelled out than when hiding it in a constant.